### DISCUSSION TITLE
如何在本地加载此插件

### DISCUSSION THREAD

**Discussion by wei-z-git** [he0119/nonebot-plugin-wordcloud#128] [Q&A]
求问如何在本地加载此插件，
目录结构：
```
/项目
  --bot.py
  --nonebot-plugin-wordcloud/nonebot_plugin_wordcloud
```
参考官网：[load_plugins](https://deploy-preview-1604--nonebot2.netlify.app/docs/tutorial/create-plugin#)使用`nonebot.load_plugins("nonebot-plugin-wordcloud/nonebot_plugin_wordcloud")`或`nonebot.load_plugin(Path(nonebot-plugin-wordcloud/nonebot_plugin_wordcloud))`
都会出现获取不了插件名的报错：
```
 File "g:\AI\projecta\.venv\lib\site-packages\nonebot_plugin_datastore\utils.py", line 34, in get_caller_plugin_name
    raise ValueError("自动获取插件名失败")
ValueError: 自动获取插件名失败
```

看了下目录结构也差不多，俺其他几个本地插件都可以通过上面方法加载，只不过没有二级目录，比如
`nonebot.load_plugin("nonebot_plugin_forwarder")`
![image](https://user-images.githubusercontent.com/32572815/224525796-661552a8-56ed-4b4e-81c2-fdc966025d56.png)

**Comment by he0119** [comment]
无法获取的原因，主要是插件的模块名出了问题。你只需要直接 `nonebot.load_plugins("nonebot-plugin-wordcloud")`。

`load_plugins` 是加载目录下的插件，`nonebot_plugin_wordcloud` 这整个包算一个插件。

**Comment by wei-z-git** [comment]
感觉和本地pip之前安的包有关，尝试卸载并使用`nonebot.load_plugins("nonebot-plugin-wordcloud")`发现另一个问题，
```
  File "G:\AI\little_dove_ai_bot\nonebot-plugin-wordcloud\nonebot_plugin_wordcloud\migrations\f77200370a39_ob12.py", line 13, in <module>
    from nonebot_plugin_wordcloud.model import Schedule
ModuleNotFoundError: No module named 'nonebot_plugin_wordcloud'
```
似乎是包路径的问题，但不知原因，凭感觉我先临时把

1. 一级目录`nonebot-plugin-wordcloud`改为` nonebot_plugin_wordcloud`
2. `from nonebot_plugin_wordcloud.model import Schedule`改为`from nonebot_plugin_wordcloud.nonebot_plugin_wordcloud.model import Schedule`
后总算导入进来了
3. `nonebot.load_plugins("nonebot-plugin-wordcloud")`
可喜可贺，可喜可贺....

**Reply by he0119** [reply]
这个我确实没注意，改成相对导入就好了。

```diff
- from nonebot_plugin_wordcloud.model import Schedule
+ from ..model import Schedule
```

**Reply by wei-z-git** [reply]
相对路径还是会报错, 或许和我用.venv虚拟环境有关(Windows经常碰到这种)，暂时我觉得可以先不管了：
```
  File "G:\AI\projecta\nonebot_plugin_wordcloud\nonebot_plugin_wordcloud\migrations\f77200370a39_ob12.py", line 13, in <module>
    from ..model import Schedule
ImportError: attempted relative import with no known parent package
```

**Reply by he0119** [reply]
嗯，那算了吧。又不是不能用.jpg

**Comment by he0119** [comment]
在 0.4.8(<https://github.com/he0119/nonebot-plugin-wordcloud/commit/257fee71b7c43ff68103c8f78bc9a6ea2f00c8c5>) 版本，修复了相对路径的报错。
